### PR TITLE
add separate flags for windows and linux in configmap for geneva logs…

### PR DESCRIPTION
**Pull request by ganga1980**
Expose configmap flag to disable Linux or Windows for the Geneva Logs Integration.

### CODE REVIEW

File: build/common/installer/scripts/tomlparser-geneva-config.rb [hunk 1]
```diff
@@ -256,27 +274,30 @@ def is_configure_geneva_env_vars()
...
-    file.write("export MONITORING_GCS_ACCOUNT=#{@geneva_account_name}\n")
-    file.write("export MONITORING_GCS_REGION=#{@geneva_gcs_region}\n")
-    file.write("export MONITORING_CONFIG_VERSION=#{@geneva_logs_config_version}\n")
-    file.write("export MONITORING_GCS_AUTH_ID=#{@geneva_gcs_authid}\n")
-    file.write("export MONITORING_GCS_AUTH_ID_TYPE=AuthMSIToken")
-  end
-  file.write("export GENEVA_LOGS_INFRA_NAMESPACES=#{@infra_namespaces}\n")
-  file.write("export GENEVA_LOGS_TENANT_NAMESPACES=#{@tenant_namespaces}\n")
+  if @disable_linux
+    puts "config::geneva_logs::info:geneva logs integration disabled for linux"
```
Comment (jatakiajanvi12) [inline build/common/installer/scripts/tomlparser-geneva-config.rb:278]: Should we have this put above itelf, where we are setting the var?

File: build/common/installer/scripts/tomlparser-geneva-config.rb [hunk 2]
```diff
@@ -256,27 +274,30 @@ def is_configure_geneva_env_vars()
...
+  else
+    file.write("export GENEVA_LOGS_INTEGRATION=#{@geneva_logs_integration}\n")
```
Comment (jatakiajanvi12) [inline build/common/installer/scripts/tomlparser-geneva-config.rb:280]: Even if disabled, should we set this as false?

File: build/common/installer/scripts/tomlparser-geneva-config.rb [hunk 3]
```diff
@@ -289,42 +310,46 @@ def is_configure_geneva_env_vars()
...
-    if is_configure_geneva_env_vars()
-      commands = get_command_windows("MONITORING_GCS_ENVIRONMENT", @geneva_account_environment)
-      file.write(commands)
-      commands = get_command_windows("MONITORING_GCS_NAMESPACE", @geneva_account_namespace_windows)
-      file.write(commands)
-      commands = get_command_windows("MONITORING_GCS_ACCOUNT", @geneva_account_name)
+    if @disable_windows
+      puts "config::geneva_logs::info:geneva logs integration disabled for windows"
+    else
+      commands = get_command_windows("GENEVA_LOGS_INTEGRATION", @geneva_logs_integration)
```
Comment (jatakiajanvi12) [inline build/common/installer/scripts/tomlparser-geneva-config.rb:316]: Same as above

File: build/common/installer/scripts/tomlparser-geneva-config.rb (hunk 2 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser-geneva-config.rb:280]: > Even if disabled, should we set this as false?


if disabled, none of the GENEVA related env var being set.

File: build/common/installer/scripts/tomlparser-geneva-config.rb (hunk 3 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser-geneva-config.rb:316]: > Same as above
if disabled, none of the GENEVA related env var being set.

File: build/common/installer/scripts/tomlparser-geneva-config.rb (hunk 1 above)
Comment (ganga1980) [reply build/common/installer/scripts/tomlparser-geneva-config.rb:278]: > Should we have this put above itelf, where we are setting the var?

as discussed, keeping the behavior of not setting any of the GENEVA env var if disabled.

**Review by jatakiajanvi12 — APPROVED** [review]

**Review by wanlonghenry — APPROVED** [review]

**Pull request merged**